### ISSUE TITLE
Neural Network Practice

### ISSUE THREAD

**Issue by jerryff** [thunlp/NLP-THU#4]
These are some very general questions, but I sincerely hopes for some guidance because of lack of experience.
When we reproduce the neural network in the paper, if there is a gap with the experimental results, what should be done next?
Take homework2 as an example. And the reference paper is:
     _Assessing State-of-the-Art Sentiment Models on State-of-the-Art Sentiment Datasets_

**Q1:** How to judge whether the gap is derived from the details of the implementation, or some bugs? Can we achieve the performance of the paper simply by tuning the parameters? 
      For example, in HW2, the paper's result for a simple LSTM model is 45.3. But I implement that model and just got 31-34 after parameters tuning. After adding a CNN layer, it reach 36-39. Can it reach 40+ simply by tuning?

**Q2:** If tuning is enough, are there any approach for parameters tuning? 
     It seems that changing those parameters mentioned in the HW2 or paper (among the specified range or use the paper's parameters) does not work.

**Q3:** If the loss and accuracy fluctuate during training process and between multiple training sessions, is the result reliable? If not, what can I do to solve this problem?
     For example, the attached graph show the training and validation loss of one training process.
![WechatIMG75](https://user-images.githubusercontent.com/18180843/56718317-1c94f400-6771-11e9-8a90-91155f7b96db.jpeg)

**Comment by EdwardZH** [comment]
For question1, it is hard to say your implement models have bugs. Some tricks and hyperparameters will affect model performance significantly. Therefore, check all hyperparameters are same as the original paper. Using some advanced optimizers, such as Adam. Check tricks in the paper are added in your implement models, such as Dropout.

**Comment by EdwardZH** [comment]
For question2, you can also tune the batch size, learning rate and hidden layer.

**Comment by EdwardZH** [comment]
For question3, the loss seems to descend all time. You can shuffle your data during training and use Adam optimizer for improvement.
